providers/oidc_test.go

```diff
@@ -373,3 +373,36 @@ func TestOIDCProvider_findVerifiedIdToken(t *testing.T) {
 	assert.Equal(t, nil, err)
 	assert.Equal(t, true, verifiedIDToken == nil)
 }
+
+func TestExtractRawgroupsFromClaim(t *testing.T) {
+	server, provider := newTestSetup([]byte(""))
+	defer server.Close()
+
+	provider.GroupsClaim = "groups"
+
+	testCases := map[string]struct {
+		RawClaims map[string]interface{}
+		Groups    []string
+	}{
+		"ArrayGroup": {
+			RawClaims: map[string]interface{}{"groups": []string{"one_group", "another_group"}},
+			Groups:    []string{"one_group", "another_group"},
+		},
+		"StringGroup": {
+			RawClaims: map[string]interface{}{"groups": "one_group"},
+			Groups:    []string{"one_group"},
+		},
+		"NumberGroup": {
+			RawClaims: map[string]interface{}{"groups": 42},
+			Groups:    []string(nil),
```

Comment:
If a number comes in the JWT claim, I would expect the string form of it to be in the session.

Reply:
I'll just remove the testcase, this case is just there to see if the method doesn't panic or blow up

Reply:
I think you should have the test case, but you need to assert for `[]string{"42"}` -- I think this is a corner case we might run into (we've been burned by IdPs not following specs many times in the past).

Reply:
now supports this corner case